src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -1186,6 +1186,42 @@ namespace UiaOperationAbstraction
             return *std::get<LocalType>(m_member);
         }
 
+        UiaBool operator==(const UiaArray& rhs) const
+        {
+            if (ShouldUseRemoteApi())
+            {
+                auto mutableThis = *this;
+                mutableThis.ToRemote();
+
+                auto mutableRhs = rhs;
+                mutableRhs.ToRemote();
+                return std::get<RemoteType>(mutableThis.m_member).IsEqual(std::get<RemoteType>(mutableRhs.m_member));
+            }
+
+            auto localVector = std::get<LocalType>(m_member);
+            auto localRhsVector = std::get<LocalType>(rhs.m_member);
+
+            if (localVector->size() != localRhsVector->size())
+            {
+                return false;
+            }
+
+            for (size_t i = 0; i < localVector->size(); ++i)
+            {
+                if (localVector->at(i) != localRhsVector->at(i))

```

Comment:
No need to use `at`, if we've validated the bounds already. `at` throws when indexing out of bounds, but we can't index out of bounds here (loop pre-conditions guarantee this), so might as well use `operator[]` that will skip those checks.